Sources/ScipioKit/DescriptionPackage.swift

```diff
@@ -99,9 +102,34 @@ struct DescriptionPackage {
 }
 
 extension DescriptionPackage {
-    func resolveBuildProducts() throws -> Set<BuildProduct> {
+    func resolveBuildProducts() throws -> [BuildProduct] {
         let targetsToBuild = try targetsToBuild()
-        return Set(try targetsToBuild.flatMap(resolveBuildProduct(from:)))
+        var products = try targetsToBuild.flatMap(resolveBuildProduct(from:))
+
+        let productMap: [String: BuildProduct] = Dictionary(products.map { ($0.target.name, $0) }) { $1 }
+        func resolvedTargetToBuildProduct(_ target: ResolvedTarget) -> BuildProduct {
+            productMap[target.name]!
+        }
```

Comment:
https://github.com/giginet/Scipio/pull/84/files#r1277289261 のコメント読みました。
異常系なのでエラーハンドリングについて同意ですが、調査しやすくするために `fatalError` は投げたいです。（起きなさそうだが）
また、force unwrapの意味も後から追いづらいため、冗長ですが、エラー文を書いておくことで、コメントして頂いたような意図を把握する上でも良いと思いました。

```suggestion
        func resolvedTargetToBuildProduct(_ target: ResolvedTarget) -> BuildProduct {
            if let buildProduct = productMap[target.name] {
                return buildProduct
            } else {
                fatalError("Unable to find \(target.name) in the build list")         
            }
        }
```

Reply:
了解です。

suggestionいただいでますが、こういうケースは `guard` にしたほうが、
その式のエラーケースについて忘れることができて望ましいと思うので、
そのように修正します。

Reply:
`guard` + `preconditionFaillure` に書き換えました。

今回のような「理屈上は起きないが何らかの手違いで起きてしまったエラー」は、
logic failureと言って `precondition` 系で表現するのが適切とされています。